specs/dotsigstore_bundle/pb/sigstore_common.proto

```diff
@@ -0,0 +1,96 @@
+// Copyright 2022 The Sigstore Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+syntax = "proto3";
+package sigstore.common;
+
+option go_package = "github.com/sigstore/proto/common";
+option java_package = "com.github.sigstore.proto.common";
+
+// This package defines commonly used message types within the Sigstore
+// community.
+
+// Only a subset of the secure hash standard algorithms are supported.
+// See https://nvlpubs.nist.gov/nistpubs/FIPS/NIST.FIPS.180-4.pdf for more
+// details.
+// UNSPECIFIED SHOULD not be used, primary reason for inclusion is to force
+// any proto JSON serialization to emit the used hash algorithm, as default
+// option is to *omit* the default value of an emum (which is the first
+// value, represented by '0'.
+enum HashAlgorithm {
+        UNSPECIFIED = 0;
+        SHA2_256 = 1;
+        SHA2_512 = 2;
+}
+
+// HashOutput captures a digest of a 'message' (generic octet sequence)
+// and the corresponding hash algorithm used.
+message HashOutput {
+        HashAlgorithm algorithm = 1;
+        // This is the raw octets of the message digest as computed by
+        // the hash algorithm.
+        bytes digest = 2;
+}
+
+// MessageSignature stores the computed signature over a message.
+message MessageSignature {
+        // Message digest can be used to identify the artifact.
+        HashOutput message_digest = 1;
+        // The raw bytes as returned from the signature algorithm.
+        // The signature algorithm (and so the format of the signature bytes)
+        // are determined by the contents of the 'verification_material',
+        // either a key-pair or a certificate. If using a certificate, the
+        // certificate contains the required information on the signature
+        // algorithm.
+        // When using a key pair, the algorithm MUST be part of the public
+        // key, which MUST be communicated out-of-band.
+        bytes signature = 2;
+}
+
+// This message holds a RFC 3161 timestamp.
+message RFC3161SignedTimestamp {
+        // Signed timestamp is the DER encoded TimeStampToken
```

Comment:
@znewman01 @haydentherapper FYI: I changed this to include the TimeStampToken instead of TSTInfo. After reading [RFC3161](https://www.ietf.org/rfc/rfc3161.html) and [RFC2630](https://datatracker.ietf.org/doc/html/rfc2630) more thoroughly I realized that TSTInfo does not capture the actual signature.

Reply:
It's actually `TimeStampResp`, not just the token

Reply:
I might be wrong cause I haven't looked into the clients much, but I think most (openssl allows either, but defaults to TimeStampResp) expect the TimeStampResp structure, even with PKIStatusInfo having little value. 

I'd probably prefer that clients don't have to extract anything from this response either and can simply persist it.

Reply:
As this is only a comment as of now (what the verifier should expect after decoding the bytes) we can be a bit flexible in the early stages. Also, I think that actually using this field is not something the early adopters will use, I would expect Rekor tlog entries for the near future. But regardless, getting the construct right early on is crucial. 

What I could find int terms of libraries: 
* [Go](https://github.com/digitorus/timestamp/blob/master/timestamp.go#L292): Both response and token.
* [Java](https://github.com/bcgit/bc-java): Both response and token
* [Python](https://github.com/trbs/rfc3161ng/blob/master/rfc3161ng/api.py#L128): Seems to be centered around the time stamp token, but offers decode functions from DER to timestamp token.
* [JavaScript](https://github.com/PeculiarVentures/asn1-schema/tree/master/packages/tsp/src): Seems to offer both methods.

If we feel that that storing the entire `TimeStampResp` is more future proof, then we should change to that. I honestly have to little cross-language experience to give any advice.